### DISCUSSION TITLE
Why doesn't WebView2 run when an unpackaged WinUI3 desktop app is run as as single instance

### DISCUSSION THREAD

**Discussion by rcohn** [MicrosoftEdge/WebView2Feedback#3962] [Q&A]
Hello -

I wanted to make sure only one instance of our WinUI3  desktop is running.  So I used the technique that sets DISABLE_XAML_GENERATED_MAIN and the method show below using DispatcherQueueSynchronizationContext.  However, when this is done, then WebView2 does not start up.  Is there a way to configure the single-instance startup that supports WebView2?

Thank you.
Robert

(Sorry about the screen shot. The code markup in this forum flattens all the whitespace.)

![image](https://user-images.githubusercontent.com/2237248/172934891-4bfefbc9-f787-4c4e-9ffa-46dda43b1926.png)

**Comment by champnic** [comment]
Thanks for the question @rcohn!

@krschau Any ideas on why this might be?

**Comment by krschau** [comment]
I'm not sure. @rcohn, could you possibly attach a min-repro app I could take a look at?

**Comment by rcohn** [comment]
Hi @krschau -

I wish I could send you a demo showing the problem.  However, I cannot provide even a vanilla demo that does **_not_** use DISABLE_XAML_GENERATED_MAIN because now WebView2 will not launch at all.  

Last week, I had no problem launching a WebView2 demo; now WebView2 is not launching its URL (Source="https://www.microsoft.com").  So, if you could give me any guidance as to why WebView2 would not launch, that would be appreciated.  And then we can get back to my original question.  

The app is built as a WinUI3 desktop unpackaged app, using WindowsAppSDK 1.0.  The demo is based on the WinUI3 Getting Started demo (https://docs.microsoft.com/en-us/microsoft-edge/webview2/get-started/winui), except I didn't need to install the Nuget Webview2 support.

Couldn't attach the entire app sources, but I did attach MainWindow.xaml and MainWindows.xaml.cs.  Please let me know if there's any other info I can provide.

Thanks for your assistance.
Robert

[MainWindow.zip](https://github.com/MicrosoftEdge/WebView2Feedback/files/8883884/MainWindow.zip)

**Comment by tylersouthard** [comment]
@krschau Here is a minimal repro. 

It appears the minimal change to make it work for this repro is to change the `static Main` entry point from the `Task`-based version to the synchronous (`void` returning) version. If that change is made in the attached repro, the web view works. Apparently there is something wrong with using the `Task`-based `Main` when using `WebView2`.

[WebView2Test.zip](https://github.com/MicrosoftEdge/WebView2Feedback/files/9598729/WebView2Test.zip)

**Reply by rcohn** [reply]
It's good to know the workaround using the synchronous Main version.  However, we punted the problem by not using DISABLE_XAML_GENERATED_MAIN.  Instead we just look for another instance of the app running from the app's OnLaunched method.  If already running, then the new instance bails out.

Thanks for your help.
